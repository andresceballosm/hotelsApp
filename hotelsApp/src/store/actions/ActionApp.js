import CONSTANTS from '../CONSTANSTS';

export const ActionSetLoading = () => ({
    type: CONSTANTS.SET_LOADING,
});

export const ActionStopLoading = () => ({
    type: CONSTANTS.STOP_LOADING,
});

export const ActionSetCity = (city) => ({
    type: CONSTANTS.SET_CITY,
    city
});