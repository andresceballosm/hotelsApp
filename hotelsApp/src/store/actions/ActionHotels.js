import CONSTANTS from '../CONSTANSTS';

export const ActionGetHotelByParameter = (parameter, value) => ({
    type: CONSTANTS.GET_HOTELS_BY_PARAMETER,
    parameter, value
});

export const ActionSetHotelByParameter = (hotels) => ({
    type: CONSTANTS.SET_HOTELS,
    hotels
});

export const ActionGetHotelByParameters = (query) => ({
    type: CONSTANTS.GET_HOTELS_BY_PARAMETERS,
    query
});



