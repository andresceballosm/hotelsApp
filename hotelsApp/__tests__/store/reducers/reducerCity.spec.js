import configureMockStore from 'redux-mock-store';
import { ActionSetCity } from '../../../src/store/actions/ActionApp';

const mockStore = configureMockStore();

describe('Testing log in reducerCity', () => {  
    const store = mockStore({});
  
    beforeEach(() => {
        store.clearActions();
    });
    
    it('set city in state', async () => {
        await store.dispatch(ActionSetCity('Bogota'));
        expect(store.getActions()).toMatchSnapshot();
    });
});
