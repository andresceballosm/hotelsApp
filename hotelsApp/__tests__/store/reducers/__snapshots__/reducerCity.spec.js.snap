// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing log in reducerCity service return hotels 1`] = `
Array [
  Object {
    "city": "Bogota",
    "type": "SET_CITY",
  },
]
`;

exports[`Testing log in reducerCity set city in state 1`] = `
Array [
  Object {
    "city": "Bogota",
    "type": "SET_CITY",
  },
]
`;
